### PR TITLE
Metrics deployment

### PR DESCRIPTION
**Pull request by Gregory-Pereira**
Partially addresses: #281 
cc @RobotSail @vishnoianil @nerdalert 

Changes:

- App of apps architecture for argo
- Umami deployment manifests for openshift and kind
    - Everything here works, but I cannot access the ingress for umami after first deploying the `kind.yaml` and the `kind-ingress.yaml` and then deploying the umami stack. That being said I tried the same thing with the Instructlab-ui kind manifests, and the ingress for that also didn't work so I assume this is a non-issue.
- Some `kind` docs on stuff I learned through struggling with it

After this PR lands, we need to deploy it, make sure its working, and then go in and configure the settings. After that is done, we can setup a sites for QA and Prod, and grab their respective tracker codes to to embed in the TS codebase. Then we simply do check on the env variable and inject the correct script tag.

### CODE REVIEW

**Review by RobotSail — APPROVED** [review]

File: .gitignore [hunk 1]
```diff
@@ -26,5 +26,10 @@ pathservice.pid
 /playwright-report/
 /blob-report/
 /playwright/.cache/
-
 playwright/.auth
+# dont track secrets in git
+deploy/k8s/overlays/kind/umami/umami-secret.yaml
+deploy/k8s/overlays/openshift/umami/umami-secret.yaml
+# for dubgging kustomize builds
+*/tmp.yaml
```
Comment (vishnoianil) [inline .gitignore:34]: want to remove these lines?

File: Makefile [hunk 2]
```diff
@@ -111,9 +115,37 @@ check-kubectl:
...
+		echo "Please install kubeseal" ; \
+		echo "https://github.com/bitnami-labs/sealed-secrets?tab=readme-ov-file#kubeseal" ; \
+		exit 1 ; \
+	fi
+
+.PHONY: check-sealed-secrets-controller
+check-sealed-secrets-controller:
+	$(CMD_PREFIX) kubectl get deployment ${SEALED_SECRETS_CONTROLLER_NAME} -n ${SEALED_SECRETS_CONTROLLER_NAMESPACE} > /dev/null 2>&1 || { \
+		echo "Error: Could not find the Sealed Secrets controller deployment named '${SEALED_SECRETS_CONTROLLER_NAME}' in namespace '${SEALED_SECRETS_CONTROLLER_NAMESPACE}'."; \
+		echo "Please update SEALED_SECRETS_CONTROLLER_NAME and SEALED_SECRETS_CONTROLLER_NAMESPACE at the top of the Makefile"; \
```
Comment (vishnoianil) [inline Makefile:130]: i think these variable needs to be in ${}?

File: deploy/k8s/overlays/openshift/umami/umami-route.yaml [hunk 3]
```diff
@@ -0,0 +1,17 @@
+apiVersion: route.openshift.io/v1
```
Comment (vishnoianil) [inline deploy/k8s/overlays/openshift/umami/umami-route.yaml:1]: I assume this route will allow use to access the umami login page using ui.instructlab.ai:3001 and qa.ui.instructlab.ai:3001?

File: docs/umami_metrics.md [hunk 4]
```diff
@@ -0,0 +1,98 @@
+# Umami metrics
```
Comment (vishnoianil) [inline docs/umami_metrics.md:1]: Thanks for the documentation, this is awesome.

File: docs/umami_metrics.md [hunk 5]
```diff
@@ -0,0 +1,98 @@
...
+
+Introduction
```
Comment (vishnoianil) [inline docs/umami_metrics.md:3]: probably some more context here?

**Review by vishnoianil — COMMENTED** [review]
awesome work @Gregory-Pereira . Few minor comments. Can you add some text around how to access the umami dashboard in kind and openshift environment?

File: Makefile (hunk 2 above)
Comment (Gregory-Pereira) [reply Makefile:130]: No, this is a logging message telling users to make sure they are set properly.

File: deploy/k8s/overlays/openshift/umami/umami-route.yaml (hunk 3 above)
Comment (Gregory-Pereira) [reply deploy/k8s/overlays/openshift/umami/umami-route.yaml:1]: So I did try to make those changes but OpenShift Routes do not directly expose services on custom ports like 3001 to external clients via the host field. To do this we either would have to swap to an ingress or proxy it. I think this is suggestion is good, but as discussed there will be multiple rounds of PRs related to the metrics deployment. Lets make an issue for it and address later.

**Comment by Gregory-Pereira** [comment]
I would consider your review addressed @vishnoianil. I did try to make the route changes you were talking about but they would require another refactor (swapping to ingress or something else). For that reason, lets keep a separate route for the umami portal disconnected from the UI domain. We can iterate as we go - I would love to eventually see it as tab in our UI.

**Review by vishnoianil — APPROVED** [review]
LGTM

**Pull request merged**